python.d/ceph.chart.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+# Description: ceph netdata python.d module
+# Author: Luis Eduardo (lets00)
+
+import rados
+import json
+from bases.FrameworkServices.SimpleService import SimpleService
+
+ORDER = ['general', 'pool_usage', 'pool_objects', 'osd_usage']
+
+CHARTS = {
+    'general': {
+        'options': [None, 'Ceph General Options', 'usage', 'general', 'ceph.general', 'line'],
+        'lines': []
+    },
+    'pool_usage': {
+        'options': [None, 'Ceph Pools', 'usage', 'pool usage', 'ceph.pool_usage', 'line'],
+        'lines': []
+    },
+    'pool_objects': {
+        'options': [None, 'Ceph Pools', 'objects', 'number of objects', 'ceph.pool_objects', 'line'],
+        'lines': []
+    },
+    'osd_usage': {
+        'options': [None, 'Ceph OSDs', 'usage', 'osd usage', 'ceph.osd_usage', 'line'],
+        'lines': []
+    }
+}
+
+PRESENTATION_MAP = {
+    'KB': 1,
+    'MB': 1024,
+    'GB': 1024**2,
+    'TB': 1024**3
+}
+
+
+class Service(SimpleService):
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.config_file = self.configuration.get('config_file')
+        self.keyring_file = self.configuration.get('keyring_file')
+        self.presentation_data = self.configuration.get('presentation_data')
+        self.presentation_value = None
+        self.cluster = rados.Rados(conffile=self.config_file,
+                                   conf=dict(keyring=self.keyring_file))
+
+    def check(self):
+        """
+        Checks module
+        :return:
+        """
+        if not (self.config_file and self.keyring_file):
+            self.error('config_file and/or keyring_file is not defined')
+            return False
+
+        # Set default presentation_data to GB if it does not defined and declare
+        # presentation_value with PRESENTATION_MAP value.
+        if not self.presentation_data:
+            self.presentation_data = 'GB'
+        self.presentation_value = PRESENTATION_MAP[self.presentation_data]
+        try:
+            self.cluster.connect()
+        except Exception as error:
+            self.error(error)
+            return False
```

Comment:
Do we need to support reconnect?
like
https://github.com/firehol/netdata/blob/master/python.d/beanstalk.chart.py#L239-L245

Reply:
I will see if this is possible.